hypothesis-python/tests/cover/test_float_nastiness.py

```diff
@@ -261,3 +261,24 @@ def test_can_exclude_neg_infinite_endpoint(x):
 @given(st.floats(1e307, float("inf"), exclude_max=True))
 def test_can_exclude_pos_infinite_endpoint(x):
     assert not math.isinf(x)
+
+
+def test_exclude_infinite_endpoint_is_invalid():
+    with pytest.raises(InvalidArgument):
+        st.floats(min_value=float("inf"), exclude_min=True).validate()
+    with pytest.raises(InvalidArgument):
+        st.floats(max_value=float("-inf"), exclude_max=True).validate()
+
+
+def test_exclude_entire_interval():
+    with pytest.raises(InvalidArgument, match="exclude_min=.+ and exclude_max="):
+        st.floats(0, 0, exclude_min=True, exclude_max=True).validate()
```

Comment:
A couple other scenarios for this that it might be worth testing explicitly:

* What happens if I call this with `st.floats(0, 0)` and *only one* of `exclude_min` or `exclude_max`?
* What happens if I call this with `st.floats(-0.0, 0.0, exclude_min=True, exclude_max=True)`?

(I imagine both work, or at least I haven't checked that they don't, they just seem like corner cases worth testing)

Reply:
If by "work" you mean "raise InvalidArgument", yep.  I also dropped a line in the docs about excluding +/- 0.0